### PR TITLE
feat(Android): sentry integration

### PR DESCRIPTION
**Pull request by KaylaBrady**
### Summary

_Ticket:_ [🤖 | Setup | Sentry integration](https://app.asana.com/0/1205732265579288/1206707047117773/f)

What is this PR for?

This PR adds the Sentry integration for the android client. 

iOS
- [ ] If you added any user-facing strings on iOS, are they included in Localizable.xcstrings?
  - [ ] Add temporary machine translations, marked "Needs Review"
android
- [ ] All user-facing strings added to strings resource~

### Testing

What testing have you done?

I ran locally (disabling the debug check) and confirmed the error was logged in the new project.
image

I don't know how to test the debug symbol upload locally, so I'm hoping to merge this PR as-is with the known temporary message capture on app launch and verify that it works as expected.

### CODE REVIEW

File: androidApp/build.gradle.kts [hunk 1]
```diff
@@ -20,81 +35,119 @@ android {
...
-    }
-    buildFeatures { compose = true }
-    packaging { resources { excludes += "/META-INF/{AL2.0,LGPL2.1}" } }
-    buildTypes { getByName("release") { isMinifyEnabled = false } }
-    flavorDimensions += "environment"
-    productFlavors {
-        create("staging") {
-            dimension = "environment"
-            applicationIdSuffix = ".staging"
+        buildFeatures {
```
Comment (boringcactus) [inline androidApp/build.gradle.kts:38]: Was it intentional to delete the closing curly brace on the `defaultConfig` block and move a ton of things inside it and most of the file into the `android` block?

Comment (KaylaBrady) [reply androidApp/build.gradle.kts:38]: 😱 it was not!

File: README.md [hunk 2]
```diff
@@ -67,6 +67,10 @@ ios app:
 
 To switch between the staging and prod app flavors, go to Build > Select Build Variant and then set the `:androidApp` Active Build Variant.
 
+Populate any configuration needed in your the .envrc file. These will be read by a gradle buld task
```
Comment (boringcactus) [inline README.md:70]:
```suggestion
Populate any configuration needed in your the .envrc file. These will be read by a gradle build task
```

**Review by boringcactus — APPROVED** [review]

**Comment by KaylaBrady** [comment]
Update: 
Removing the temporary log. Debug files were uploaded as part of our CI build. We probably don't need to be doing that, but it doesn't seem to increase run times in a meaningful way, so seems simplest to leave as-is.
image

**Pull request merged**